core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/CromwellSwaggerClient.scala

```diff
@@ -22,8 +22,13 @@ class CromwellSwaggerClient(cromwellBasePath: String) extends LazyLogging {
   }
 
 
-  def describe(userInfo: UserInfo, wdl: String): Try[WorkflowDescription] = {
-    Retry.retry(5.seconds, 30.seconds) { Try { getCromwellWomtoolApi(userInfo.accessToken.token).describe("v1", wdl, null, null, null, null) } }
-  }
+  def describe(userInfo: UserInfo, wdl: WDL): Try[WorkflowDescription] =
+    wdl.url match {
+      case Some(url) =>
+        Retry.retry(5.seconds, 30.seconds) { Try { getCromwellWomtoolApi(userInfo.accessToken.token).describe("v1", null, url, null, null, null) } }
+      case None =>
+        Retry.retry(5.seconds, 30.seconds) { Try { getCromwellWomtoolApi(userInfo.accessToken.token).describe("v1", wdl.source, null, null, null, null) } }
+    }
```

Comment:
Reading this, it seems like a `WDL` is an "or" relation versus an "and" relation...

WdlSource OR WdlUrl, not WdlSource AND WdlUrl?

Thus would a sealed trait make more sense than a case class / Product?

Reply:
See comment in `WDL.scala`, they're not designed to be exclusive, rather to coexist and be used in complement with each other.
```
 // The kernel of Cromwell's `cromwell.core.WorkflowSourceFilesCollection` - a workflow and its location, to enable relative imports
```

Reply:
Just kidding, your idea proved extremely good